src/main/java/dk/kb/datahandler/kaltura/KalturaDeltaUploadJob.java

```diff
@@ -0,0 +1,304 @@
+package dk.kb.datahandler.kaltura;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.Http2SolrClient;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.kaltura.client.enums.MediaType;
+
+import dk.kb.datahandler.config.ServiceConfig;
+import dk.kb.kaltura.client.DsKalturaClient;
+import dk.kb.storage.model.v1.StreamErrorTypeDto;
+import dk.kb.storage.util.DsStorageClient;
+import dk.kb.util.webservice.exception.InternalServiceException;
+
+/**
+ * This class has a single public method to upload all kaltura streams with from mTime (in Solr) and higher values. 
+ */
+public class KalturaDeltaUploadJob {
+
+    private static DsKalturaClient kalturaClient = null;
+    private static final Logger log = LoggerFactory.getLogger(KalturaDeltaUploadJob.class);
+
+   
+    /**
+    * <p>
+    * Start job that will upload missing streams to kaltura.
+    * <p>
+    * Workflow:
+    * <p>
+    * 1) Extract records from Solr using mTimeFrom and with the condition
+    * access_malfunction:false AND production_code_allowed:true AND NOT
+    * kaltura_id:* Only extract the few fields from solr that are required:
+    * title,description,file_id,id,resource_description,originates_from 
+    * 
+    * <p>
+    * 2) Calculate full path to the stream. Calculation depend on if record is DOMS or Preservica.
+    *    Validate file exists and not too short bytesize. If this happens mark the record with error and skip.
+    * <p>
+    *  3) Check that the file_id has not been uploaded to kaltura before. If it has use the kaltura internal id for this record.
+    * <p>   
+    * 4) For each new stream:
+    * <p>
+    *  4.1) Upload the stream to kaltura. (Notice some streams do not have extension, but this seems not to be an issue with kaltura). The kaltura tag
+    *  for upload is 'delta-2025-05-01' where last part is current day.
+    * <p>
+    *  4.2) Update the record's kalturaid in ds-storage 
+    * <p>
+    * All records that has been updated with error or kalturaId will have mTime updated to new value.
+    * After completion the facade method will start a solr delta-index job. 
+    * 
+    * @param mTimeFrom Upload all streams for records in solr with mTimeFrom higher than this value. 
+    * @throws InternalServiceException If any Solr or Kaltura call fails. Stop uploading more. Maybe allow single kaltura upload jobs to fail later.
+    * 
+    */
+    
```

Comment:
Please remove the linebreak between javadoc and method

Reply:
Done.